about/src/main/java/dev/arkbuilders/components/about/QRCryptoDialog.kt

```diff
@@ -0,0 +1,255 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.VerticalDivider
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+
+@Composable
+internal fun QRCryptoDialog(
+    visible: Boolean,
+    title: String,
+    wallet: String,
+    fileName: String,
+    @DrawableRes qrBitmap: Int,
+    onDismiss: () -> Unit
+) {
+    if (visible.not())
+        return
+
+    Dialog(
+        onDismissRequest = onDismiss,
+        properties = DialogProperties(usePlatformDefaultWidth = false)
+    ) {
+        Content(
+            title = title,
+            wallet = wallet,
+            fileName = fileName,
+            qrBitmap = qrBitmap,
+            onDismiss = onDismiss
+        )
+    }
+}
+
+@SuppressLint("ResourceType")
+@Composable
+private fun Content(
+    title: String,
+    wallet: String,
+    fileName: String,
+    @DrawableRes qrBitmap: Int,
+    onDismiss: () -> Unit
+) {
+    val ctx = LocalContext.current
+    val clipboardManager = LocalClipboardManager.current
+    val launcher =
+        rememberLauncherForActivityResult(ActivityResultContracts.CreateDocument("image/jpg")) { uri ->
+            uri ?: return@rememberLauncherForActivityResult
+            val input = ctx.resources.openRawResource(qrBitmap)
+            ctx.contentResolver.openOutputStream(uri).use { output ->
+                output?.let {
+                    input.copyTo(output)
+                }
```

Comment:
File IO operations should be done in worker thread rather than main thread for performance reason.
Also, `output` stream has to be closed after copy completion as well.

Reply:
`use` automatically closes the stream. But yes, i don't need to do this in main thread.